operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -33,6 +33,7 @@ public abstract class AbstractDependentResource<R, P extends HasMetadata>
   private final boolean updatable = this instanceof Updater;
   private final boolean deletable = this instanceof Deleter;
   private final DependentResourceReconciler<R, P> dependentResourceReconciler;
+  private final ThreadLocal<R> desiredCache = new ThreadLocal<>();
```

Comment:
Not sure if using ThreadLocal is the best for this purpose, if some later call the dependent from reconcile method it will happen on different thread so desired won't be available

Reply:
Yeah, I'm not sure either if that's the proper way to do it…